internal/gatewayserver/hooks/requestid.go

```diff
@@ -0,0 +1,45 @@
+package hooks
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/rs/xid"
+	"github.com/twitchtv/twirp"
+
+	"github.com/razorpay/trino-gateway/internal/constants/contextkeys"
+)
+
+const (
+	requestIDHttpHeaderKey = "X-Request-ID"
+)
+
+// RequestID returns function which puts unique request id into context.
+func RequestID() *twirp.ServerHooks {
+	hooks := &twirp.ServerHooks{}
+
+	hooks.RequestRouted = func(ctx context.Context) (context.Context, error) {
+		//var err error
+		requestID, _ := ctx.Value(contextkeys.RequestID).(string)
+		if requestID == "" {
+			requestID = xid.New().String()
+		}
+		ctx = context.WithValue(ctx, contextkeys.RequestID, requestID)
+		return ctx, nil
+	}
+
+	return hooks
+}
+
+// WithRequestID is a http handler which puts specific http request header into
+// request context which is made available in twirp hooks.
+// Refer: https://twitchtv.github.io/twirp/docs/headers.html
+func WithRequestID(h http.Handler) http.Handler {
```

Comment:
@utk-spartan Don't seem to find it's usage in the code. When will this be called?